pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -275,12 +264,16 @@ func (w *PodWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.
 
 	allErrs := jobframework.ValidateJobOnUpdate(oldPod, newPod)
 	allErrs = append(allErrs, validateCommon(newPod)...)
-
-	allErrs = append(allErrs, validation.ValidateImmutableField(podGroupName(newPod.pod), podGroupName(oldPod.pod), groupNameLabelPath)...)
```

Comment:
Keep the check it is immutable, but only check if the oldPod has the `GroupNameLabel` specified. So, that we allow to set it, but not mutate. It cannot also be taken back.

Reply:
Done